src/cryptoadvance/specter/notifications/notification_manager.py

```diff
@@ -0,0 +1,407 @@
+              │  NotificationManager  │ ├─────────────────────► │ JSConsoleNotifications│
+              │   .create_and_show    │ │                       │                       │
+              └───────────────────────┘ │                       └───────────────────────┘
+                                        │                       ┌───────────────────────┐
+                                        └──────────────────────►│   JSNotifications     │
+                                                                └───────────────────────┘
+
+    2. internal_notification  at the example of on_close
+
+                  ┌───────────────────────┐
+                  │                       │  if "internal_notification" in notification.target_uis:
+                  │  NotificationManager  │ ──────────────────────────────────────►┐
+                  │   .create_and_show    │                                        │
+                  └───────────────────────┘                                        ▼
+                                                                 ┌────────────────────────────┐
+                                                                 │  NotificationManager       │
+                                                                 │   ._treat_internal_message │
+                                                                 │     e.g. on_close          │
+                                                                 └─────────────────┬──────────┘
+                                                                                   │ on_close
+                                                                                   ▼
+                   ┌───────────────────────┐                         ┌───────────────────────┐
+                   │   JSNotifications     │ ◄─────────────────────┐ │   NotificationManager │
+                   │   .on_close           │                       │ │   ._internal_on_close │
+                   └───────────────────────┘                       │ └───────────────────────┘
+                   ┌───────────────────────┐                       │
+                   │  NotificationManager  │                       │
+                   │  ._delete_notification│ ◄─────────────────────┘
+                   └───────────────────────┘
+
+    """
+
+    def __init__(self, websockets_port, user_manager, ui_notifications=None):
+        """
+        Arguments:
+            - ui_notifications:  {user_id: [list of ui_notifications]}
+                    The "default" ui_notifications is at position 0
+
+            websockets_port = -1 disables the websocket part
+        """
+        self.ui_notifications = ui_notifications if ui_notifications else []
+        self.notifications = []
+        self._register_default_ui_notifications()
+
+        self.websockets_server = None
+        self.websockets_client = None
+        if websockets_port != -1:
```

Comment:
Can we do that explicitely with a second value in `config.py`? Magic values are a bad practice and it's not clear what there meaning is.

Reply:
Yes you are right, I overlooked that, when  finally solving the pytest issue. I will add `SPECTER_WEBSOCKET_ACTIVE`

Reply:
Done